frontend/src/metabase-lib/queries/drills/underlying-records-drill.js

```diff
@@ -13,12 +13,14 @@ export function underlyingRecordsDrill({ question, clicked }) {
     return null;
   }
 
-  // the metric value should be the number of rows that will be displayed
-  const rowCount = typeof clicked.value === "number" ? clicked.value : 2;
+  // here we expect rows count of an aggregated query item, but actually we have numeric metric value, so
+  // we don't really know number of rows in original un-aggregated data
+  const value = typeof clicked.value === "number" ? clicked.value : 2;
+
   const tableName = query.table() && query.table().displayName();
 
   return {
-    rowCount,
+    value: value < 0 ? 2 : value,
```

Comment:
It feels the logic is scattered. There is a logic to get the value from `clicked` and set default value if it's not present, then there is another logic to normalize the negative value and set it to 2 if it's negative. These 2 logic live in different place when I feel it should be at the same place. Also to conditionally pass a variable `value` as a `value` key looks weird. It means the `value` on line 18 isn't necessarily the same value as line 23.

What do you think about extracting this logic to another function so we could simplify the code here

considering this pseudo code
```ts
const value = getValue(clicked)
...
return {
  value,
  ...
```